test/gtest/ucp/test_ucp_fence.cc

```diff
@@ -145,18 +146,31 @@ class test_ucp_fence : public test_ucp_atomic {
             receiver().connect(&sender());
         }
 
-        memheap = malloc(memheap_size);
-
         params.field_mask = UCP_MEM_MAP_PARAM_FIELD_ADDRESS |
                             UCP_MEM_MAP_PARAM_FIELD_LENGTH |
                             UCP_MEM_MAP_PARAM_FIELD_FLAGS;
-        params.address    = memheap;
         params.length     = memheap_size;
         params.flags      = GetParam().variant;
+        if (params.flags & UCP_MEM_MAP_FIXED) {
+            params.address  = (void *)(uintptr_t)0xFF000000;
+            params.flags   |= UCP_MEM_MAP_ALLOCATE;
+        } else {
+            memheap = malloc(memheap_size);
+            params.address = memheap;
+            params.flags = params.flags & (~UCP_MEM_MAP_ALLOCATE);
+        }
 
         status = ucp_mem_map(receiver().ucph(), &params, &memh);
         ASSERT_UCS_OK(status);
 
+        status = ucp_mem_query(memh, &mem_attr);
+        ASSERT_UCS_OK(status);
+        EXPECT_TRUE(mem_attr.field_mask & (UCP_MEM_ATTR_FIELD_ADDRESS |
+                                           UCP_MEM_ATTR_FIELD_LENGTH));
+        EXPECT_LE(memheap_size, mem_attr.length);
+        if (!memheap) {
```

Comment:
Because if `memheap` allocated by `malloc` then `mem_attr.address` may different in common case, I'm not sure that the test will work fine everywhere :) Are you suggesting to remove the check?